pubsub/pubsub.go

```diff
@@ -637,13 +637,15 @@ func (s *Subscription) getNextBatch(nMessages int) ([]*driver.Message, error) {
 
 	g, ctx := errgroup.WithContext(s.backgroundCtx)
 	for _, maxMessagesInBatch := range batches {
+		// Make a copy of the loop variable since it will be used by a goroutine.
```

Comment:
ah, the famous [gorouting using loop iterator variable](https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables). I think `go vet` detects these, but not if one uses `group.Go` (just plain goroutines)